pkg/sdk/testint/tasks_gen_integration_test.go

```diff
@@ -338,120 +344,180 @@ func TestInt_Tasks(t *testing.T) {
 	})
 
 	t.Run("clone task: default", func(t *testing.T) {
-		sourceTask := createTask(t)
+		sourceTask, taskCleanup := testClientHelper().Task.Create(t)
+		t.Cleanup(taskCleanup)
 
 		id := testClientHelper().Ids.RandomSchemaObjectIdentifier()
-
-		request := sdk.NewCloneTaskRequest(id, sourceTask.ID())
-
-		err := client.Tasks.Clone(ctx, request)
+		err := client.Tasks.Clone(ctx, sdk.NewCloneTaskRequest(id, sourceTask.ID()))
 		require.NoError(t, err)
-		t.Cleanup(cleanupTaskProvider(id))
+		t.Cleanup(testClientHelper().Task.DropTaskFunc(t, id))
 
 		task, err := client.Tasks.ShowByID(ctx, id)
 		require.NoError(t, err)
 
-		assertTask(t, task, request.GetName())
+		assertTask(t, task, id, testClientHelper().Ids.WarehouseId().Name())
```

Comment:
It's not clear why we check warehouse id here. Shouldn't we just compare if some values are identical in these 2 tasks?

Reply:
Hmm true, refactored the test case a bit